src/HttpServer/Response.gren

```diff
@@ -0,0 +1,67 @@
+module HttpServer.Response exposing
+    ( Response
+    , toResponse
+    , write
+    , header
+    , status
+    , end
+    )
+
+
+import Dict exposing (Dict)
+import Task exposing (Task)
+
+
+type Response =
+    Response KernelResponse
+        { status : Int
+        , headers : Dict String String
+        , body : Array String -- TODO: support bytes
+        }
+
+
+type KernelResponse =
+    -- Note: Actual implementation in Kernel code
+    KernelResponse
+
+
+toResponse : KernelResponse -> Response
+toResponse response =
+    Response response
+        { status = 200
+        , body = []
+        , headers = Dict.empty
+        }
+
+
+status : Int -> Response -> Response
+status statusCode (Response kernelResponse response) =
+    Response kernelResponse { response | status = statusCode }
+
+
+header : String -> String -> Response -> Response
+header key value (Response kernelResponse response) =
+    Response kernelResponse
+        { response 
+            | headers = Dict.insert key value response.headers
+        }
+
+
+write : String -> Response -> Response
+write chunk (Response kernelResponse response) =
+    -- TODO: should this be a task?
```

Comment:
My thinking is that if it's a task, it could be performed and chained in update at any point, and that could take the place of `.end` as I have it now. Not sure I like that API though.

Another alternative is making body a single string instead of an array to make it match the semantics of the other setters (set a single value instead of appending) and renaming to `.body`. I originally had it like that, but then I was worried about the future possibilities sending large binary data. But now that I say that, perhaps that should be a different function, so something like...

```elm
Response.body data -- set the entire body of the response, not a task
Response.write data -- task that responds with a chunk of data when performed, but does not end the response
```

If that's the case, maybe the non-task functions should be prefixed with `set` (`setStatus`, `setBody`, etc...)?

Reply:
I guess there's still value in chunking it into a queue that performs at the end though. But queue feels weird to me. What about:

```elm
|> Response.status 200
|> Response.chunk "a"
|> Response.chunk "b"
|> Response.chunk "c"
|> Response.send
```
would be the same as:

```elm
|> Response.status 200
|> Response.body "abc"
|> Response.send
```

Reply:
...except in the latter case, you end up with one `response.write` in node instead of three.

Reply:
Hmm...

Thinking about this some more. Wouldn't it be likely that receiving and respond to requests would be seperated from the rest of your business logic? So, perhaps it isn't really all that important to be able to `chunk` things, as people would most often set the entire body all at once.

Chunking, might be a better fit for a stream-oriented api (which doesn't have to be a part of this first version).

Reply:
I also think that `setStatus` and `setBody` are better names. While it's clear that `Response.status 200` sets the status, I mostly associate those names with getters.

Reply:
All good points. If it's all setters though now I'm back to thinking it should just be one "respondWith" function that takes a record...

Reply:
Hmm...

I think the nice thing about the current API is that you can rely on defaults. For instance, in the future we might have a `Respond.sendJson`, which can set both status and header to 200 and application/json for you, in which case you might not have to worry about the status or header fields at all.